src/prometheus/definition.go

```diff
@@ -701,3 +724,41 @@ func getRawEntityID(parentGroupLabel, groupLabel, entityID string, groups defini
 	}
 	return rawEntityID, nil
 }
+
+func WithLabelValueInMetricNameAndWithLabelValueMappedToMetricValue(rawMetricName, label, format, valueLabel string, valueMap map[string]int) definition.FetchFunc {
+	return func(groupLabel, entityID string, rawGroups definition.RawGroups) (definition.FetchedValue, error) {
+		fetchedValues := definition.FetchedValues{}
+		metricsRaw, err := definition.FromRaw(rawMetricName)(groupLabel, entityID, rawGroups)
+		if err != nil {
+			return nil, fmt.Errorf("fetching metric %q for entity %q in group %q: %w", rawMetricName, entityID, groupLabel, err)
+		}
+
+		metrics, ok := metricsRaw.([]Metric)
+		if !ok {
+			return nil, fmt.Errorf("incompatible metric type for %q. Expected: []Metric. Got: %T", rawMetricName, metricsRaw)
+		}
+
+		for _, metric := range metrics {
+			nameLabelValue, ok := metric.Labels[label]
+			if !ok {
+				return nil, fmt.Errorf("label %q for metric name not found in metric %v", label, metric)
+			}
+
+			fetchedValueName := fmt.Sprintf(format, nameLabelValue)
+
+			valueLabelValue, ok := metric.Labels[valueLabel]
```

Comment:
Not very happy about the naming. If this is the intermediate value, what about
```suggestion
			originalMetricValue, ok := metric.Labels[valueLabel]
```

Reply:
Yeah, name is not great. Maybe `labelValue`? Or `requestedLabelValue`? I wanted to indicate it's a label value, which will become a metric value. Or `metricValueFromLabel`? Or `fetchedValuefromLabel`?

Reply:
`fetchedValuefromLabel` +1

Reply:
Changed.